onnxscript/rewriter/pattern_test.py

```diff
@@ -780,6 +780,119 @@ def test_model1(x: FLOAT[16, 32], y: FLOAT[16, 32]) -> FLOAT[16, 32]:
         self.assertEqual([x.op_type for x in model.graph], ["ReluPlus"])
 
 
+class ValueNodeCheckersTest(unittest.TestCase):
+    """Test value/node level checkers functionality."""
+
+    def test_pattern_match_with_node_checker(self):
+        """Test Pattern.match with node-level checker."""
+        def add_node_checker(context, node):
+            return node.op_type == "Add"
+        
+        # Create a pattern that matches Add operations with a node checker
+        def add_pattern(op, x, y):
+            return op.Add(x, y, _check=add_node_checker)
+        
+        # Create the pattern
+        rule_pattern = pattern.Pattern(add_pattern)
+        
+        # Create a simple model
+        model_proto = onnx.parser.parse_model(
+            """
+            <ir_version: 7, opset_import: [ "" : 17]>
+            agraph (float[N] x, float[N] y) => (float[N] z)
+            {
+                z = Add(x, y)
+            }
+            """
+        )
+        model = ir.serde.deserialize_model(model_proto)
+        
+        # Find the Add node in the model
+        nodes = list(model.graph.all_nodes())
+        add_node = nodes[0]
+        self.assertEqual(add_node.op_type, "Add")
+        
+        # Try to match the pattern
+        match_result = rule_pattern.match(model, model.graph, add_node)
+        
+        self.assertIsNotNone(match_result)
+        self.assertEqual(len(match_result.nodes), 1)
+        self.assertEqual(len(match_result.node_bindings), 1)
+
+    def test_pattern_match_with_failing_node_checker(self):
+        """Test Pattern.match with failing node-level checker."""
+        def failing_node_checker(context, node):
+            return False  # Always fail
+        
+        # Create a pattern that matches Add operations with a failing node checker
+        def add_pattern(op, x, y):
+            return op.Add(x, y, _check=failing_node_checker)
+        
+        # Create the pattern
+        rule_pattern = pattern.Pattern(add_pattern)
+        
+        # Create a simple model
+        model_proto = onnx.parser.parse_model(
+            """
+            <ir_version: 7, opset_import: [ "" : 17]>
+            agraph (float[N] x, float[N] y) => (float[N] z)
+            {
+                z = Add(x, y)
+            }
+            """
+        )
+        model = ir.serde.deserialize_model(model_proto)
+        
+        # Find the Add node in the model
+        nodes = list(model.graph.all_nodes())
+        add_node = nodes[0]
+        self.assertEqual(add_node.op_type, "Add")
+        
+        # Try to match the pattern
+        match_result = rule_pattern.match(model, model.graph, add_node)
+        
+        # Should fail due to failing checker
+        self.assertIsNone(match_result)
+
+    def test_pattern_match_with_value_checker(self):
+        """Test Pattern.match with value-level checker."""
+        def value_checker(context, value):
```

Comment:
Change this checker to `is_positive_constant`. In the implementation of this check, check if value has a const_value, and check if that const_value (a numpy array) represents a single value that is positive.

In the test model below, have several calls to Add: one with first parameter non-constant, one with first parameter a constant (initializer) that has a positive value, and one with negative value. And check that all cases are handled correctly.